yt_idefix/fields.py

```diff
@@ -47,3 +55,137 @@ def setup_fluid_fields(self):
+                    * (data[("gas", "pressure")] / data[("gas", "density")])
+                )
+
+            self.add_field(
+                ("gas", "temperature"),
+                sampling_type="cell",
+                function=_temp,
+                units=unit_system["temperature"],
+            )
+
+        if (self.ds._dataset_type, "Temp") in self.field_list:
+            self.add_output_field(
+                (self.ds._dataset_type, "Temp"),
+                sampling_type="cell",
+                units=unit_system["temperature"],
+            )
+            self.alias(
+                ("gas", "temperature"),
+                (self.ds._dataset_type, "Temp"),
+                units=unit_system["temperature"],
+            )
+        elif (self.ds._dataset_type, "Temperature") in self.field_list:
+            self.add_output_field(
+                (self.ds._dataset_type, "Temperature"),
+                sampling_type="cell",
+                units=unit_system["temperature"],
+            )
+            self.alias(
+                ("gas", "temperature"),
+                (self.ds._dataset_type, "Temperature"),
+                units=unit_system["temperature"],
+            )
+
+        # Add sound speed; Mach number is automatically calculated by yt!
+        if hasattr(self.ds, "gamma"):
+
+            def _sound_speed(field, data):
+                return np.sqrt(
+                    data.ds.gamma
+                    * (data[("gas", "pressure")] / data[("gas", "density")])
+                )
+
+            self.add_field(
+                ("gas", "sound_speed"),
+                sampling_type="cell",
+                function=_sound_speed,
+                units=unit_system["velocity"],
+            )
+
+        if (self.ds._dataset_type, "ndens") not in self.field_list and hasattr(
+            self.ds, "mu"
+        ):
+
+            def _ndens(field, data):
+                return data["gas", "density"] / (data.ds.mu * mh)
+
+            self.add_field(
+                ("gas", "number_density"),
+                sampling_type="cell",
+                function=_ndens,
+                units=unit_system["length"] ** -3,
+            )
+        # Optional particle number density field in PLUTO
+        if (self.ds._dataset_type, "ndens") in self.field_list:
+            self.add_output_field(
+                (self.ds._dataset_type, "ndens"),
+                sampling_type="cell",
+                units=unit_system["length"] ** -3,
+            )
+            self.alias(
+                ("gas", "number_density"),
+                (self.ds._dataset_type, "ndens"),
+                units=unit_system["length"] ** -3,
+            )
+
+    def setup_particle_fields(self, ptype):
+        super().setup_particle_fields(ptype)
+        # This will get called for every particle type.
+        # {TODO} Starting with version 4.4 PLUTO has particle support and this function needs an implementation in future
```

Comment:
Todo comments should start with `TODO:` or `TODO(name):`. I use text editor extensions to discover them so they need to be standardised

Reply:
I'll correct this after the history rewrite! Sorry I have somehow missed this!